docs/new-architecture-app-modules-android.md

```diff
@@ -0,0 +1,425 @@
+        into("$buildDir/react-ndk/exported")
+    }
+
+    afterEvaluate {
+        preBuild.dependsOn(packageReactNdkLibs)
+        configureNdkBuildDebug.dependsOn(preBuild)
+        configureNdkBuildRelease.dependsOn(preBuild)
+    }
+
+    packagingOptions {
+        pickFirst '**/libhermes.so'
+        pickFirst '**/libjsc.so'
+    }
+}
+```
+
+Finally, we need to create a Makefile inside the `src/main/jni` folder called `Android.mk` with the following content:
+
+```makefile
+THIS_DIR := $(call my-dir)
+
+include $(REACT_ANDROID_DIR)/Android-prebuilt.mk
+include $(GENERATED_SRC_DIR)/codegen/jni/Android.mk
+
+include $(CLEAR_VARS)
+
+LOCAL_PATH := $(THIS_DIR)
+LOCAL_MODULE := myapplication_appmodules
+
+LOCAL_C_INCLUDES := $(LOCAL_PATH) $(GENERATED_SRC_DIR)/codegen/jni
+LOCAL_SRC_FILES := $(wildcard $(LOCAL_PATH)/*.cpp) $(wildcard $(GENERATED_SRC_DIR)/codegen/jni/*.cpp)
+LOCAL_EXPORT_C_INCLUDES := $(LOCAL_PATH) $(GENERATED_SRC_DIR)/codegen/jni
+
+# Please note as one of the library listed is libreact_codegen_samplelibrary
+# This name will be generated as libreact_codegen_<library-name>
+# where <library-name> is the one you specified in the Gradle configuration
+LOCAL_SHARED_LIBRARIES := libjsi libfbjni libglog libfolly_json libyoga libreact_nativemodule_core libturbomodulejsijni librrc_view libreact_render_core libreact_render_graphics libfabricjni libfolly_futures libreact_debug libreact_render_componentregistry libreact_render_debug libruntimeexecutor libreact_codegen_rncore libreact_codegen_samplelibrary
+
+LOCAL_CFLAGS := \
+    -DLOG_TAG=\"ReactNative\"
+LOCAL_CFLAGS += -fexceptions -frtti -std=c++17 -Wall
+
+include $(BUILD_SHARED_LIBRARY)
+```
+
+This setup will run a native build on your project and will compile the C++ files that have been generated by the codegen. You will see the native build running with the Gradle task `:app:externalNativeBuildRelease`
+
+You can now verify that everything works correctly by running your android app:
+
+```bash
+yarn react-native run-android
+```
+
+## 2. Java - Provide a `ReactPackageTurboModuleManagerDelegate`
+
+Now is time to actually use the TurboModule.
+First, we will need to create a `ReactPackageTurboModuleManagerDelegate` subclass, like the following:
+
+```java
+package com.awesomeproject;
+
+import com.facebook.jni.HybridData;
+import com.facebook.react.ReactPackage;
+import com.facebook.react.ReactPackageTurboModuleManagerDelegate;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.soloader.SoLoader;
+
+import java.util.List;
+
+public class MyApplicationTurboModuleManagerDelegate extends ReactPackageTurboModuleManagerDelegate {
+
+    private static volatile boolean sIsSoLibraryLoaded;
+
+    protected MyApplicationTurboModuleManagerDelegate(ReactApplicationContext reactApplicationContext, List<ReactPackage> packages) {
+        super(reactApplicationContext, packages);
+    }
+
+    protected native HybridData initHybrid();
+
```

Comment:
here we need to add this line
`native boolean canCreateTurboModule(String moduleName);`
otherwise, I am getting this error in logcat
```
W/dererboilerpat: CheckJNI: method to register "canCreateTurboModule" not in the given class. This is slow, consider changing your RegisterNatives calls.
E/dererboilerpat: ----- class 'Lcom/reactnativeturbomodulefabricrendererboilerpate/MyApplicationTurboModuleManagerDelegate;' cl=0x13cc2420 -----
E/dererboilerpat:   objectSize=265 (120 from super)
E/dererboilerpat:   access=0x0008.0001
E/dererboilerpat:   super='java.lang.Class<com.facebook.react.ReactPackageTurboModuleManagerDelegate>' (cl=0x13cc2420)
E/dererboilerpat:   vtable (2 entries, 3 in super):
E/dererboilerpat:      0: com.facebook.jni.HybridData com.reactnativeturbomodulefabricrendererboilerpate.MyApplicationTurboModuleManagerDelegate.initHybrid()
E/dererboilerpat:      1: void com.reactnativeturbomodulefabricrendererboilerpate.MyApplicationTurboModuleManagerDelegate.maybeLoadOtherSoLibraries()
E/dererboilerpat:   direct methods (1 entries):
E/dererboilerpat:      0: void com.reactnativeturbomodulefabricrendererboilerpate.MyApplicationTurboModuleManagerDelegate.<init>(com.facebook.react.bridge.ReactApplicationContext, java.util.List)
E/dererboilerpat:   static fields (1 entries):
E/dererboilerpat:      0: boolean com.reactnativeturbomodulefabricrendererboilerpate.MyApplicationTurboModuleManagerDelegate.sIsSoLibraryLoaded
E/dererboilerpat: 
E/dererboilerpat: Failed to register native method com.reactnativeturbomodulefabricrendererboilerpate.MyApplicationTurboModuleManagerDelegate.canCreateTurboModule(Ljava/lang/String;)Z in /data/app/com.reactnativeturbomodulefabricrendererboilerpate-P3yaQha3eGo-5UIRn2clnw==/base.apk!classes19.dex
E/log: error java.lang.NoSuchMethodError: no static or non-static method "Lcom/reactnativeturbomodulefabricrendererboilerpate/MyApplicationTurboModuleManagerDelegate;.canCreateTurboModule(Ljava/lang/String;)Z"
A/dererboilerpat: java_vm_ext.cc:542] JNI DETECTED ERROR IN APPLICATION: JNI NewGlobalRef called with pending exception java.lang.NoSuchMethodError: no static or non-static method "Lcom/reactnativeturbomodulefabricrendererboilerpate/MyApplicationTurboModuleManagerDelegate;.canCreateTurboModule(Ljava/lang/String;)Z"
A/dererboilerpat: java_vm_ext.cc:542] (Throwable with no stack trace)
A/dererboilerpat: java_vm_ext.cc:542] 
A/dererboilerpat: java_vm_ext.cc:542]     in call to NewGlobalRef
A/dererboilerpat: java_vm_ext.cc:542]     from java.lang.String java.lang.Runtime.nativeLoad(java.lang.String, java.lang.ClassLoader)
A/dererboilerpat: java_vm_ext.cc:542] "create_react_context" prio=5 tid=28 Runnable
```

Reply:
Thanks for the heads up. Yeah I was investigating this one. @RSNara Can you confirm that `canCreateTurboModule` is a required method for a TM?

Reply:
I've just cross checked once more: `canCreateTurboModule` is not necessary, and you should be good to go building without it.